Extensions/TweenBehavior/tweenruntimebehavior.js

```diff
@@ -86,23 +122,39 @@ gdjs.TweenRuntimeBehavior.prototype._removeObjectFromScene = function (
   return this.owner.deleteFromScene(this._runtimeScene);
 };
 
-gdjs.TweenRuntimeBehavior.prototype._removeTween = function (identifier) {
+gdjs.TweenRuntimeBehavior.prototype._removeTween = function(identifier) {
   this._tweens[identifier].instance.stop();
   delete this._tweens[identifier];
 };
 
-gdjs.TweenRuntimeBehavior.prototype._setupTweenEnding = function (
+gdjs.TweenRuntimeBehavior.prototype._setupTweenEnding = function(
   identifier,
   destroyObjectWhenFinished
 ) {
   if (destroyObjectWhenFinished) {
-    this._tweens[identifier].instance.tween().then(() => {
-      this._removeObjectFromScene(identifier);
-    });
+    this._tweens[identifier].instance
+      .tween()
+      .catch(() => {
```

Comment:
Shouldn't we turn this into an ES5 function?

Reply:
Right, we should indeed.
Note that for arrow functions using `this`, `this` is not the same in a `function`, so they won't work the same.
The solution is to declare a variable: `var that = this;` and then use `that` in the `function`:

```
gdjs.TweenRuntimeBehavior.prototype._setupTweenEnding = function(...) {
  var that = this;
  this._tweens[identifier].instance
  .tween()
  .catch(function() {
    ...
  }).then(function() {
    that.xxx = ...
    that.yyy();
  })
}
```

it's a bit of pain to remember to do this, hence why arrow functions are super useful.